### ISSUE TITLE
Configure Pop-up: Insert Link and Image Buttons

### ISSUE THREAD

**Issue by wspiking** [cmndrbensisko/LocalLayer#7]
The Insert Link and Insert Image buttons in the Configure Pop-up window don't seem to be working.  Please verify.

**Comment by WashGIS** [comment]
you beat me too it. It is not working for me either

**Comment by rscheitlin** [comment]
This issue will be fixed in the next release

**Comment by wspiking** [comment]
Thank you Robert.  One other note on the pop-up config.   In the Custom pop-up config window,  is there a way to add <p></p> when a user adds a hard return?

**Comment by rscheitlin** [comment]
Currently just type &lt;br&gt;. But that is something I will look at.

**Comment by WashGIS** [comment]
Thank you

**Comment by wspiking** [comment]
Hi Robert,  I'm still having issues w/ the Insert Link and Insert Image buttons.  Also, the hard return is just being added to the first line.  See attached image.  To make sure I reinstalled WAB Dev and loaded it up with LocalLayer 1.3 and it didn't make a difference.  Let me know if you need me to do any additional testing.
Thank you sir,
Bill
![capture](https://cloud.githubusercontent.com/assets/5851986/6043524/60da7736-ac52-11e4-9afb-5db4cc6306e3.PNG)

**Comment by rscheitlin** [comment]
William, Can you verify that you copied the the new version of the widget to your stemapp widgets folder and the widget folder of any app that you have already built and are attempting to use the Local Layer widget in. You can check that you actually using the new version by opening the manifest.json and check the version number in that file. If you replaced the widget in the stemapp but not in your built apps widget directory then your built app will be using the old version still. Another why right off you can tell that you are using the 1.3 vs the 1.2 is there will be a Add Tiled Layer option in the widget.

**Comment by wspiking** [comment]
I'll verify this morning and get back to you.  Question....since wab dev runs in a web browser, should a clear my cache/history between updates or changes in the code?  I remember seeing the Add Tile Layer option, but I didn't test.  Maybe browser caching was the issue?

**Comment by wspiking** [comment]
Hi Robert, I'm still seeing the same result.  It might be good if someone else tested as well.  Here are my steps.
1. Cleared my browsers history/cache
2. Started w/ a new instance of wab dev
3. Re-installed LocalLayer 1.3 in wab dev
4. Tested in Chrome and IE 10

![capture](https://cloud.githubusercontent.com/assets/5851986/6061261/bcf3f60a-ad0b-11e4-8956-c0451cd291c0.PNG)

**Comment by rscheitlin** [comment]
Bill,
  OK, I have read your comment closer now. "Also, the hard return is just being added to the first line" the &lt;br&gt; that you are seeing in the first line is likely one that you manually added in the text. The widget converts the TeaxtArea dijits \n (hard breaks) behind the scenes into &lt;br&gt;, you will not see this in the popup. When you say the insert image and hyperlink do not work can you go into more specific detail?

**Comment by wspiking** [comment]
In the example above, I'm not adding the <br> in the first line.  It is automatically being added when I hit return.....so the updates you made in 1.3 seem to be working only for the first line of text.  The Insert Link and Insert Image buttons do nothing for me when clicked. I've now tested on two different machines with the same result.

**Comment by rscheitlin** [comment]
Bill,

   you are saying that when you click the button you do not get a popup at all? Can you hit F12 key and tell me what the console is saying? 

:bulb: I would love to hear from someone else if they are seeing this as thus far I can not reproduce what you are seeing.

There is **nothing** in the code that would insert the &lt;br&gt; in the text window, so I don't understand what you are seeing concerning the first line...

**Comment by wspiking** [comment]
I'm not getting a pop-up at all.

![capture](https://cloud.githubusercontent.com/assets/5851986/6062946/43b9e96c-ad19-11e4-8df4-953c97f76d99.PNG)

**Comment by rscheitlin** [comment]
Bill, are you clicking the popup edit button before you have entered a url for the layer and it gets a little green check mark beside the url? Do you see a list of fields in the popups Key Value tab?

**Comment by wspiking** [comment]
Nope.  I'm entering the url and getting the green check mark and the service loads fine within the map.  I think we really need to find someone else to test, I don't want to waste your time if it's an issue w/ my setup.

**Comment by rscheitlin** [comment]
Bill, I can get the same exact error and the buttons not doing anything, but only if the url for the layer has not been entered and/or verified before I click the popup edit button (pencil).

**Comment by rscheitlin** [comment]
Bill, 
1. What layer type are you trying? 
2. Are you adding a new layer or are you editing an existing? 
3. Do you see anything different if you are adding a new layer versus editing and existing?

**Comment by WashGIS** [comment]
I am still having issues with the insert link and image buttons also.  Since I only had a test map in WAB builder any way I got rid of everything and started new. New install of web app builder, new download of Locallayer widget. New map.
­Results
1. ```
     Added Rest service as tiled service, got green check, clicked configure popup on sublayer, clicked customized, clicked insert link button -> nothing happened, clicked add image -> nothing happened. Opened developer tools. [cid:image001.png@01D04143.0D288D00]
   ```
2. ```
     Tried same steps adding as dynamic service ( even though it is a tiled service) same result.
   ```

**Comment by wspiking** [comment]
Thank you for testing WashGIS.

Robert, I've got to get a deliverable out this afternoon so I won't be able to continue testing until this evening.  Thanks again for your help.

**Comment by rscheitlin** [comment]
Thanks, guys I can now see the issue and am reopening the bug

**Comment by WashGIS** [comment]
So I played around with adding multiple lines in the popup.  I wish I could export my chrome dev log (if you know how let me know) here is a screen shot of the logs. [cid:image001.png@01D04145.CFA8A7F0]

I did a series something like this
1. ```
     Add dynamic layer to map
   ```
2. ```
     Click ok
   ```
3. ```
     Open config back up
   ```
4. ```
     Open popup config on sublayer
   ```
5. ```
     Add text  to custom tab (I didn’t add any html tags just used the return key on the keyboard)
   ```
6. ```
     Click ok on popup window
   ```
7. ```
     Click ok on layer config and close widget config
   ```
8. ```
     Clicked on map and popup displayed just as I wanted – only added two lines and forgot that I wanted to test with more than two lines
   ```
9. ```
     Opened back up to popup config
   ```
10.   Decided to work from empty popup so deleted everything from custom
11.   Ok out and click on map for popup
12.   Still displays fields I added to the custom popup
13.   Reopend popup config
14.   Popup config shows fields checked on (I didn’t check anything)
15.   Removed checked fields
16.   Ok out and click on map for popup
17.   Empty. Good!
18.   Reopend popoup config
19.   Add text  to custom tab (I didn’t add any html tags just used the return key on the keyboard) (this time four lines)
20.   Ok out and click on map for popup
21.   Displayed as expected
22.   Reopend popup config
23.    “widget” reformatted an added <br> for the first return
24.   Clicked ok to close popup config
25.   Reopend popup config and next return has been replace with <br>

It was something like that.  Hope it helps and was not too hard to follow.

From: Robert J Scheitlin [mailto:notifications@github.com]
Sent: Thursday, February 05, 2015 9:56 AM
To: cmndrbensisko/LocalLayer
Cc: Rathvon, Joe
Subject: Re: [LocalLayer] Configure Pop-up: Insert Link and Image Buttons (#7)

Bill,

you are saying that when you click the button you do not get a popup at all? Can you hit F12 key and tell me what the console is saying?

[https://assets-cdn.github.com/images/icons/emoji/unicode/1f4a1.png]I would love to hear from someone else if they are seeing this as thus far I can not reproduce what you are seeing.

There is nothing in the code that would insert the <br> in the text window, so I don't understand what you are seeing concerning the first line...

—
Reply to this email directly or view it on GitHubhttps://github.com/cmndrbensisko/LocalLayer/issues/7#issuecomment-73058877.

**Comment by rscheitlin** [comment]
WashGIS,
   Those details are extremely helpful in diagnosing the issues I have in the GUI. Thanks
